### PR TITLE
Php functions

### PR DESCRIPTION
**Pull request by alex-dyak**

### CODE REVIEW

File: tests/SayHelloArgumentTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    protected function setUp(): void
+    {
+        $this->functions = new functions\Functions();
+    }
+
+    /**
+     * @dataProvider positiveDataProvider
+     */
+    public function testPositive($input, $expected)
```
Comment (e-ZAV) [inline tests/SayHelloArgumentTest.php]: what does **testPositive** mean?

File: tests/SayHelloArgumentTest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        $this->assertEquals($expected, $this->functions->sayHelloArgument($input));
+    }
+
+    public function positiveDataProvider(): array
```
Comment (e-ZAV) [inline tests/SayHelloArgumentTest.php]: what does **positiveDataProvider** mean?

File: tests/SayHelloArgumentTest.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        return [
+            [10, 'Hello 10'],
+            ['Alex', 'Hello Alex'],
+            ['cat', 'Hello cat'],
+            [true, 'Hello 1'],
+            [false, 'Hello '],
```
Comment (e-ZAV) [inline tests/SayHelloArgumentTest.php]: Is there any way to avoid the word "Hello" here?

File: tests/SayHelloArgumentWrapperTest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    protected function setUp(): void
+    {
+        $this->functions = new functions\Functions();
+    }
+
+    /**
+     * @dataProvider negativeDataProvider
+     */
+    public function testNegative($arg, $expected)
```
Comment (e-ZAV) [inline tests/SayHelloArgumentWrapperTest.php]: what does **testNegative** mean?

File: tests/SayHelloArgumentWrapperTest.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        $this->assertEquals($expected,$this->functions->sayHelloArgumentWrapper($arg));
+    }
+
+    public function negativeDataProvider(): array
```
Comment (e-ZAV) [inline tests/SayHelloArgumentWrapperTest.php]: what does **negativeDataProvider** mean?

File: tests/SayHelloTest.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+class SayHelloTest extends TestCase
+{
+    protected $functions;
+
+    protected function setUp(): void
+    {
+        $this->functions = new functions\Functions();
+    }
+
+    public function testPositive()
```
Comment (e-ZAV) [inline tests/SayHelloTest.php]: what does **testPositive** mean?

**Comment by e-ZAV** [comment]
I recommend creating a "**functions/**" directory for tests and moving all tests for functions there.

File: tests/functions/SayHelloArgumentTest.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+        return [
+            [10, $hello . ' 10'],
+            ['Alex', $hello . ' Alex'],
+            ['cat', $hello . ' cat'],
+            [true, $hello . ' 1'],
+            [false, $hello . ' '],
+        ];
+    }
+}
```
Comment (e-ZAV) [inline tests/functions/SayHelloArgumentTest.php]: actually, the point was to move 'Hello' word here
`$this->assertEquals('Hello ' . $expected, sayHelloArgument($input));`

**Review by e-ZAV — APPROVED** [review]

File: tests/functions/SayHelloTest.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+class SayHelloTest extends TestCase
+{
+    protected $functions;
+
+    protected function setUp(): void
+    {
+        $this->functions = new functions\Functions();
+    }
+
+    public function testPositive()
```
Comment (e-ZAV) [inline tests/functions/SayHelloTest.php]: `testPositive()`?

**Review by e-ZAV — APPROVED** [review]

File: tests/functions/SayHelloArgumentTest.php (hunk 7 above)
Comment (alex-dyak) [reply tests/functions/SayHelloArgumentTest.php]: Changed code

File: tests/functions/SayHelloTest.php (hunk 8 above)
Comment (alex-dyak) [reply tests/functions/SayHelloTest.php]: Changed test name

**Pull request merged**